packages/theia-integration/src/browser/diagram/glsp-diagram-widget.ts

```diff
@@ -49,6 +52,11 @@ export class GLSPDiagramWidget extends DiagramWidget implements SaveableSource {
     protected options: DiagramWidgetOptions & GLSPWidgetOptions;
     protected requestModelOptions: Args;
 
+    protected readonly onWidgetClosedEmitter = new Emitter<undefined>();
+    get onWidgetClosed(): Event<undefined> {
```

Comment:
Looks like this effectively does nothing. The `onWidgetClosedEmitter` never fires so classes that listen `onWidgetClosed` will never get notified.

Reply:
Thanks for spotting that, I forgot to remove that from a previous version. I moved that logic to `onCloseRequest`, so I remove the emitter after all.